zipline/pipeline/loaders/earnings_estimates.py

```diff
@@ -403,34 +393,40 @@ def load_adjusted_array(self, columns, dates, assets, mask):
         # To optimize performance, only work below on assets that are
         # actually in the raw data.
         assets_with_data = set(assets) & set(self.estimates[SID_FIELD_NAME])
+        last_per_qtr, stacked_last_per_qtr = self.get_last_data_per_qtr(
+            assets_with_data,
+            columns,
+            dates
+        )
+        # Determine which quarter is immediately next/previous for each
+        # date.
+        zeroth_quarter_idx = self.get_zeroth_quarter_idx(stacked_last_per_qtr)
+        zero_qtr_data = stacked_last_per_qtr.loc[zeroth_quarter_idx]
+
         for num_quarters, columns in groups.items():
-            last_per_qtr, stacked_last_per_qtr = self.get_last_data_per_qtr(
-                assets_with_data, columns, dates
-            )
-            # Determine which quarter is immediately next/previous for each
-            # date.
-            zeroth_quarter_idx = self.get_zeroth_quarter_idx(
-                num_quarters, stacked_last_per_qtr
-            )
-            zero_qtr_data = stacked_last_per_qtr.loc[zeroth_quarter_idx]
-            # Doing it this way because creating a MultiIndex from scratch
-            # results in being unable to unstack sids because of duplicate
-            # values, even though the MultiIndex is created with the same
-            # exact values as below - possible pandas bug.
-            requested_qtr_idx = zero_qtr_data.reset_index(
-                NORMALIZED_QUARTERS
-            ).set_index(
-                self.get_shifted_qtrs(
-                    zeroth_quarter_idx.get_level_values(NORMALIZED_QUARTERS),
-                    num_quarters
-                ),
-                append=True
-            ).index
-            requested_qtr_idx = requested_qtr_idx.rename(
-                SHIFTED_NORMALIZED_QTRS, -1
+            zero_qtr_data_idx = zero_qtr_data.index
+            requested_qtr_idx = pd.MultiIndex.from_arrays(
```

Comment:
Thinking that maybe we should move this into `get_requested_quarter_data()`, since it's not needed anywhere else, to simplify `load_adjusted_array` a bit further.